disperser/common/v2/blobstore/dynamo_metadata_store.go

```diff
@@ -476,6 +480,67 @@ func (s *BlobMetadataStore) GetBlobMetadataByAccountID(
 	return blobs, nil
 }
 
+// UpdateAccount updates the Account partition to track account activity.
+// This method performs an upsert operation, creating or updating an entry for the given account
+// with the current timestamp.
+func (s *BlobMetadataStore) UpdateAccount(ctx context.Context, accountID gethcommon.Address, timestamp uint64) error {
+	s.logger.Debug("updating account", "accountID", accountID.Hex(), "timestamp", timestamp)
+
+	item := commondynamodb.Item{
+		"PK":          &types.AttributeValueMemberS{Value: accountPK},
+		"SK":          &types.AttributeValueMemberS{Value: accountID.Hex()},
+		"Address":     &types.AttributeValueMemberS{Value: accountID.Hex()},
+		"UpdatedAt":   &types.AttributeValueMemberN{Value: strconv.FormatUint(timestamp, 10)},
+		"AccountType": &types.AttributeValueMemberS{Value: accountIndexPK},
+	}
+
+	err := s.dynamoDBClient.PutItem(ctx, s.tableName, item)
+	if err != nil {
+		return fmt.Errorf("failed to update account for accountID %s: %w", accountID.Hex(), err)
+	}
+
+	return nil
+}
+
+// GetAccounts returns accounts within the specified lookback period (newest first)
+func (s *BlobMetadataStore) GetAccounts(ctx context.Context, lookbackSeconds uint64) ([]*v2.Account, error) {
+	s.logger.Debug("querying accounts", "lookbackSeconds", lookbackSeconds)
+
+	// Calculate the cutoff timestamp
+	now := uint64(time.Now().Unix())
+	cutoffTime := now - lookbackSeconds
+
+	// Query the AccountUpdatedAtIndex GSI with time filter
+	// All account records have AccountType = "AccountIndex" which allows us to query
```

Comment:
What's the difference between AccountType and PK? Like couldn't we also just check that PK = "Account"?

Reply:
AccountType is a constant used to enable GSI on `(PK:AccountType, SK:UpdatedAt)` so the GSI only uses 1 partition.

Reply:
Renamed `AccountType` to `AccountIndex` for hopefully better context.

> Like couldn't we also just check that PK = "Account"?
Yea we could do a table scan, but we'd have to sort in post.